js-runtime/index.js

```diff
@@ -66,7 +78,19 @@ const hashv = arr => {
       }
     }
   }
-  return Number(BigInt.asIntN(64, hasher.digest())) // TODO: Move all i64 to BigInt?
+  return BigInt.asIntN(64, hasher.digest())
+}
+
+const copyarr = a => {
+  try {
+    return JSON.parse(JSON.stringify(a))
+  } catch (e) {
+    if (typeof a[0] === 'bigint') {
+      return a.map(v => BigInt(v))
+    } else {
+      return a.map(v => copyarr(v))
+    }
+  }
 }
```

Comment:
*Please* let me know if you can think of a better way. To deep clone in Javascript it's *recommended* to use JSON.parse/JSON.stringify even though that fails silently for Date objects and crashes on BigInt. This is my solution to that problem, but I know it's going to kill JS perf any time you need to clone an array of data.

Reply:
ugh

Reply:
I need to think about that. The fewer deps the better, imo. Though I did confirm that it does work on BigInt even though it doesn't have any special case code for them. Not sure what magic they're doing there.